test/dependency-graph.js

```diff
@@ -151,3 +151,92 @@ describe("dependency-graph: Affected Variables", function () {
     expect(affectedVars.length).to.equal(7);
   });
 });
+
+describe("dependency-graph: Cyclic Dependency", function() {
+
+  it('Should handle checks with no cyclic dependency', () => {
+    const expression = `
+    {% assign x = a | plus: t %}
+    {% assign y = a | times: t %}   
+    {% assign z = t | times: 3 %}
+    {% assign z = p | times: 3 %}
+    {% assign p = q | times: 3 %}
+    {% assign q = r | times: x %}
+    `;
+    const graph = depGraph.createDependencyTree(expression);
+    const cycle = depGraph.checkForCyclicDependency(graph);
+    expect(cycle).to.deep.equal([]);
+  });
+  
+  it('Should handle simple cyclic dependency check', () => {
+    // First line ceates cyclicity in the following expression
+    const expression = `
+    {% assign x = a | plus: z %}
+    {% assign y = a | times: t %}   
+    {% assign z = t | times: 3 %}
+    {% assign z = p | times: 3 %}
+    {% assign p = q | times: 3 %}
+    {% assign q = r | times: x %}
+    `;
+    const graph = depGraph.createDependencyTree(expression);
+    const cycle = depGraph.checkForCyclicDependency(graph);
+    expect(cycle).to.deep.equal([
+      'a',
```

Comment:
this is a bit weird...

it seems that `a` has a circular dependency but its actually `x` that has it?

shouldn't the execution just stop when the first `x` is encountered?

Reply:
maybe clarify this a bit in the comment describing the `checkForCyclicDependency` function?

Reply:
I'm returning the entire dfs path now. I think it'll make more sense to return the cyclic part of it only.

Also, dependency graph is of the format `baseVar -> [ vars affected by baseVar ]`, thats why the cycle is sort of reverse of what you'd expect.

Reversing cycle array just before returning should do the trick.

Reply:
Just to confirm, here the expected cycle is `[ x, z, p, q, x ]` right ?